shared/src/main/scala/mlscript/Typer.scala

```diff
@@ -145,8 +159,19 @@ class Typer(var dbg: Boolean, var verbose: Bool, var explainErrors: Bool) extend
       case _ => Set.empty // TODO TupleType?
     }
   
+  def targsMapOf(ty: Type)(implicit ctx: Ctx, raise: Raise, lookup: Option[(Str, Option[Str])] = N,
+      vars: Map[Str, SimpleType] = Map.empty): Map[Str, SimpleType] = ty match {
+    case Inter(l, r) => targsMapOf(l) ++ targsMapOf(r)
```

Comment:
Is it intended that keys in the output of `targsMapOf(l)` may be overridden by `targsMapOf(r) `? Map's `++` is a little dangerous because of this.

Reply:
Same as above. This can be changed to a nested map to support traits.

Reply:
Addressed in 6f66ec2b249323177e94b61b3cfc5e8d8dc43015